pkg/apis/extensions/validation/validation.go

```diff
@@ -659,7 +659,7 @@ func ValidatePodSecurityPolicySpec(spec *extensions.PodSecurityPolicySpec, fldPa
 	allErrs = append(allErrs, validatePSPSELinux(fldPath.Child("seLinux"), &spec.SELinux)...)
 	allErrs = append(allErrs, validatePSPSupplementalGroup(fldPath.Child("supplementalGroups"), &spec.SupplementalGroups)...)
 	allErrs = append(allErrs, validatePSPFSGroup(fldPath.Child("fsGroup"), &spec.FSGroup)...)
-	allErrs = append(allErrs, validatePodSecurityPolicyVolumes(fldPath, spec.Volumes)...)
+	allErrs = append(allErrs, validatePodSecurityPolicyVolumes(fldPath, spec.Volumes, spec.AllowedFlexVolumes)...)
```

Comment:
keep in a separate function, like `validatePSPAllowedHostPaths`?

Reply:
validate `spec.AllowedFlexVolumes` will depend on `spec.Volumes`, so I think this can reduce some code duplication.

Reply:
> validate spec.AllowedFlexVolumes will depend on spec.Volumes

not really. it can be validated separately, just like AllowedHostPaths.